### PR TITLE
Updating BYOB doc for cross-cloud and S3-compatible connectivity

### PR DESCRIPTION
**Pull request by abhinavg6**
## Description

Update the BYOB doc to add specifics for connectivity to cross-cloud or S3-compatible storage buckets.

## Ticket

NA

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [X] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [X] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [X] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

**Review by ngrayluna — APPROVED** [review]

**Comment by ngrayluna** [comment]
Note for @ngrayluna merge and deploy on Thursday 06/27

File: docs/guides/hosting/data-security/secure-storage-connector.md [hunk 1]
```diff
@@ -23,23 +23,61 @@ You can configure your bucket at both the instance level and separately for one
...
+To enable cross-cloud or S3 compatible storage, specify the storage bucket path in a pertinent format using the `GORILLA_SUPPORTED_FILE_STORES` environment variable for your W&B instance.
+
+<details>
+<summary>Connect to S3-compatible storage</summary>
+
+Specify the path using the following format:
+```text
+s3://<accessKey>:<secretAccessKey>@<url_endpoint>/<bucketName>?region=<region>
+```
+Where `region` parameter is mandatory, except for when your W&B instance is in AWS and the `AWS_REGION` environment variable is configured on the instance nodes.
```
Comment (levinandrew) [inline docs/guides/hosting/data-security/secure-storage-connector.md]: in AWS case, `AWS_REGION` could be something like us-west-2 b/c that's where their deployment is, but they their minio buckets might be set to us-east-1 (minio default region)... in that case, they would still need to use the `region` param, or update their minio bucket configuration... so maybe word like this?

"Where `region` must match the configured region of the S3-compatible bucket. This parameter is mandatory when your W&B instance is in GCP or Azure. When your W&B instance is in AWS, you can optionally omit this parameter if the `AWS_REGION` environment variable is configured on the instance nodes, and that `AWS_REGION` value also matches the configured region of the s3-compatible bucket."

File: docs/guides/hosting/data-security/secure-storage-connector.md [hunk 2]
```diff
@@ -23,23 +23,61 @@ You can configure your bucket at both the instance level and separately for one
...
+| Dedicated Cloud | X | X | Both the instance and team level BYOB are available for Amazon Web Services, Google Cloud Platform and Microsoft Azure. For the team-level BYOB, you can connect to a cloud-native storage bucket in the same or another cloud, or even a S3-compatible secure storage like [MinIO](https://github.com/minio/minio) hosted in your cloud or on-prem infrastructure. |
+| SaaS Cloud | Not Applicable | X | The team level BYOB is available only for Amazon Web Services and Google Cloud Platform. W&B fully manages the default and only storage bucket for Microsoft Azure. |
+| Self-managed | X | X | Instance level BYOB is the default since the instance is fully managed by you. If your self-managed instance is in cloud, you can connect to a cloud-native storage bucket in the same or another cloud for the team-level BYOB. You can also use S3-compatible secure storage like [MinIO](https://github.com/minio/minio) for either of instance or team-level BYOB. |
+
```
Comment (levinandrew) [inline docs/guides/hosting/data-security/secure-storage-connector.md:37]: ignorable - just cause this is a general byob doc and we are only listing the cross-cloud and on-prem connection methods, maybe we could add a paragraph like this before

"There are different ways to connect to buckets depending on the scope and deployment. For example, when connecting from a Dedicated W&B instance in one cloud to a bucket in that same cloud, you can use that cloud's native security primitives. Or, when connecting from the GCP SaaS instance to an AWS bucket, you can use GCP workload identity and assume an AWS IAM role.

However, for team-level BYOB when connecting cross-cloud or to S3-compatible storage, you can use account keys and secret keys. See below."

**Comment by abhinavg6** [comment]
@ngrayluna - I made some minor changes based on @levinandrew 's feedback, and then had to do a bit of conflict merging with your changes and a few others that had been pushed to main by this time. Do you think it's ok to merge this now? I'm ok with merging now since we've done the Dedicated Cloud deployment for the current server release today. Thanks.

**Review by levinandrew — APPROVED** [review]

**Pull request merged**